src/clj/puppetlabs/services/puppet_admin/puppet_admin_core.clj

```diff
@@ -49,33 +50,17 @@
       (jruby-puppet/mark-environment-expired! jruby-service env-name)
       (jruby-puppet/mark-all-environments-expired! jruby-service))))
 
-(defresource jruby-pool-resource
+(defn handle-jruby-pool-flush
   [jruby-service]
-  :allowed-methods [:delete]
-
-  ;; If you need to define :available-media-types, see comment below.
-  ;:available-media-types ...
-
-  :handle-exception liberator-utils/exception-handler
-
-  ;; This next line of code tells liberator to ignore any media-types
-  ;; the client has asked for.  This is necessary for this endpoint to work
-  ;; when the client sends an 'Accept: */*' header, due to the somewhat strange
-  ;; fact that this endpoint defines no 'available-media-types', since it always
-  ;; returns a '204 No Content' on success.
-  ;;
-  ;; If this resource is ever updated to define ':available-media-types' and
-  ;; return a response body, this line of code should be deleted.
-
-  :media-type-available? true
-
-  ;; Never return a '201 Created', we're not creating anything
-  :new? false
-
-  :delete!
-  (fn [context]
-    (jruby-puppet/flush-jruby-pool! jruby-service)))
-
+  (sling/try+
+    (jruby-puppet/flush-jruby-pool! jruby-service)
+    ; 204 No Content on success
+    {:status 204}
+    ; If a lock timeout occurs return a 503 Service Unavailable
+    (catch [:kind :puppetlabs.services.jruby-pool-manager.impl.jruby-agents/jruby-lock-timeout] e
```

Comment:
When I run this locally, a timeout doesn't cause anything to be written to the Puppet Server log beyond just the "Draining and refilling" message.  I think it would be good to at least log some sort of error in the case of a flush timeout so it's easy to see from looking at the log when the flush failed.  WDYT?